plasmapy/diagnostics/charged_particle_radiography.py

```diff
@@ -1115,118 +1263,161 @@ def max_deflection(self):
+        ``(hbins, vbins)``.  The default is ``(200, 200)``.
 
-        # If ignore_grid is True, use the predicted positions in the absence of
-        # simulated fields
-        if ignore_grid:
-            x = self.x0
-        else:
-            x = self.x
+    ignore_grid: `bool`
+        If `True`, returns the intensity in the image plane in the absence
+        of simulated fields.
 
-        # Determine locations of points in the detector plane using unit
-        # vectors
-        xloc = np.dot(x - self.detector, self.det_hdir)
-        yloc = np.dot(x - self.detector, self.det_vdir)
-
-        if size is None:
-            # If a detector size is not given, choose lengths based on the
-            # dimensions of the grid
-            w = self.mag * np.max(
-                [
-                    np.max(np.abs(self.grid.pts0.to(u.m).value)),
-                    np.max(np.abs(self.grid.pts1.to(u.m).value)),
-                    np.max(np.abs(self.grid.pts2.to(u.m).value)),
-                ]
-            )
+    optical_density: `bool`
+        If `True`, return the optical density rather than the intensity
+
+        .. math::
+            OD = -log_{10}(Intensity/I_0)
 
-            # The factor of 5 here is somewhat arbitrary: we just want a
-            # region a few times bigger than the image of the grid on the
-            # detector, since particles could be deflected out
-            size = 5 * np.array([[-w, w], [-w, w]]) * u.m
+        where :math:`Intensity` is the simulation intensity on the
+        detector plane and :math:`I_0` is the intensity on the detector
+        plane in the absence of simulated fields. Default is `False`.
 
-        # Generate the histogram
-        intensity, h, v = np.histogram2d(
-            xloc, yloc, range=size.to(u.m).value, bins=bins
+        If the `Intensity` histogram contains zeros, `OD` will contain
+        -`~numpy.inf` values. These can be easily replaced with zeros
+        if desired for plotting using `~numpy.nan_to_num`
+
+        ``OD = np.nan_to_num(OD, neginf=0)``
+
+    Returns
+    -------
+    hax : `~astropy.units.Quantity` array shape ``(hbins,)``
+        The horizontal axis of the synthetic radiograph in meters.
+
+    vax : `~astropy.units.Quantity` array shape ``(vbins, )``
+        The vertical axis of the synthetic radiograph in meters.
+
+    intensity : `~numpy.ndarray`, shape ``(hbins, vbins)``
+        The number of particles counted in each bin of the histogram.
+    """
+
+    # results_dict will raise an error if the simulation has not been run.
+    if isinstance(obj, Tracker):
+        d = obj.results_dict
+
+    # Check if dictionary-like (dict or npz file)
+    elif isinstance(obj, dict):
+        d = obj
+    else:
+        raise ValueError(
+            "The first argument of synthetic_radiograph must be "
+            "either a cpr.Tracker or an "
+            "output dictionary from cpr.Tracker"
```

Comment:
```suggestion
    # condition `obj` input
    if isinstance(obj, Tracker):
        # results_dict raises an error if the simulation has not been run.
        d = obj.results_dict
    elif isinstance(obj, dict):
        d = obj
    else:
        raise TypeError(
            f"Expected type dict or {Tracker} for argument `obj`, but "
            f"got type {type(obj)}."
```

This is really a `TypeError` because once you get to the else part of the clause you know `obj` is not the correct type.